File: posjava4-api/src/main/java/br/edu/utfpr/pb/posjava4api/service/GeneroService.java
package br.edu.utfpr.pb.posjava4api.service;

import java.util.List;

import br.edu.utfpr.pb.posjava4api.model.Genero;

public interface GeneroService 
			extends CrudService<Genero, Long>{

	List<Genero> findByNomeContains(String nome);
}

File: ionic/cliente-ionic/src/app/service/genero.service.ts
import { Injectable } from '@angular/core';
import { Genero } from '../model/genero';
import { HttpClient } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { CrudService } from './crud.service';

@Injectable({
  providedIn: 'root'
})
export class GeneroService extends CrudService<Genero, number> {

  constructor(http: HttpClient) {
    super(environment.api_serie + '/genero', http);
  }

}
